curve25519-dalek/src/scalar.rs

```diff
@@ -1321,6 +1321,22 @@ impl PrimeField for Scalar {
     };
 }
 
+#[cfg(feature = "group")]
+impl PrimeFieldBits for Scalar {
+  type ReprBits = [u8; 32];
+
+  fn to_le_bits(&self) -> FieldBits<Self::ReprBits> {
+    self.to_repr().into()
+  }
+
+  fn char_le_bits() -> FieldBits<Self::ReprBits> {
+    let mut bytes = (Scalar::ZERO - Scalar::ONE).to_repr();
+    bytes[0] += 1;
+    debug_assert_eq!(DScalar::from_bytes_mod_order(bytes), DScalar::ZERO);
+    bytes.into()
```

Comment:
I think this should work but haven't tested it

```suggestion
    crate::constants::BASEPOINT_ORDER.into()
```

Reply:
I'll test it now, thanks.

Reply:
That's a Scalar, not a byte array. Doesn't that break safety to have an unreduced Scalar in system? Especially now that Scalar addition assumes only unreduced Scalars exist?

Reply:
to_bytes().into() works, will push that.

Reply:
> Doesn't that break safety to have an unreduced Scalar in system?

Yeah, seems a bit odd to have it in the public API if it's unreduced /cc @rozbb

Reply:
`BASEPOINT_ORDER` will be marked deprecated from 5.0.0 - this will be `BASEPOINT_ORDER_PRIVATE` as pub(crate)
- https://github.com/dalek-cryptography/curve25519-dalek/pull/581